### ISSUE TITLE
    Are the functions of these two functions the same? 

### ISSUE THREAD

**Issue by longmalongma** [hkchengrex/STCN#33]
Are the functions of these two functions the same? They all save the current frame as a memory frame?
self.mem_k = torch.cat([self.mem_k, key], 2)；mem_out = torch.cat([mem, qv], dim=1)

![image](https://user-images.githubusercontent.com/40291722/127812003-f5a9c2de-3aeb-4c29-832f-a23d847570be.png)

![image](https://user-images.githubusercontent.com/40291722/127812047-40bc5cf8-e20b-4c73-b96b-5c97ee540772.png)

**Comment by hkchengrex** [comment]
No. The first one is a skip-connection bypassing the affinity while the second one is memorization as you mentioned.

**Comment by longmalongma** [comment]
> The first one

The first one refers to the first picture, right? Why bypass the affinity? I don't understand this.

**Comment by hkchengrex** [comment]
`mem_out` is the feature from the memory bank read using the affinity matrix. `qv` is the feature from the query frame. Since we are reconstructing the mask for the query frame, `qv` is used as a skip-connection, independent of the memory.

**Comment by longmalongma** [comment]
> No. The first one is a skip-connection bypassing the affinity while the second one is memorization as you mentioned.

Ok, I got it, thanks.
